### PR TITLE
[URL] Split forbidden host/domain code points and add more tests for each

### PR DESCRIPTION
**Pull request by karwa**

### CODE REVIEW

**Comment by annevk** [comment]
@domenic can you verify these with whatwg-url? From a cursory look this seems good.

**Comment by domenic** [comment]
It's not clear to me this is actually the direction we want to go per https://github.com/whatwg/url/pull/685#issuecomment-1021458543, so I'll hold off on further reviews until we get that clarified. It feels like we jumped to the PRs stage earlier than makes sense.

**Review by domenic — APPROVED** [review]
LGTM, will let @annevk push the merge button when he's ready to merge https://github.com/whatwg/url/pull/685

**Pull request merged**
